mp/src/vgui2/vgui_controls/CvarTextEntry.cpp

```diff
@@ -180,6 +181,18 @@ bool CvarTextEntry::HasBeenModifiedExternally() const
     return m_cvarRef.IsValid() && stricmp(m_cvarRef.GetString(), m_pszStartValue) != 0;
 }
 
+bool CvarTextEntry::ShouldUpdate(const char *szText)
+{
+    int strLength = strnlen(szText, MAX_CVAR_TEXT);
+    bool isEqToZero = atof(szText) == 0.0f,               // current text value is 0, 0.0, 0.00, etc
+         lastNumIsZero = szText[strLength - 1] == '0',    // last number is zero -> hasn't finished inputting
+         isEndDot = szText[strLength - 1] == '.',         // dont reset if end is a dot (inputing decimal)
+         isJustZero = szText[0] == '0' && strLength == 1, // 1 character and it's 0
+         resetOnZero = m_fLowestPossibleVal == 0.0f;      // if just a 0, dont reset if lowest value is >0
+
+    return (resetOnZero && isJustZero) || !((lastNumIsZero && isEqToZero) || isEndDot);
```

Comment:
I would just early return these checks as they happen, way more readable that way:

```
bool isEndDot = szText[strLength - 1] = '.';
if (isEndDot)
    return true;

etc
```